### DISCUSSION TITLE
Specify a Validator

### DISCUSSION THREAD

**Discussion by TrailBlazor** [Blazored/FluentValidation#96] [Q&A]
Hi,
Is it possible to specify the Validator to use? The reason is I want different validators for the same model.

I have the following:
```razor
<FluentValidationValidator DisableAssemblyScanning=true Validator=facilityValidator />
```
and in the code behind I have:
```csharp
protected NewFacilityValidator facilityValidator;
````
The validator is:
```csharp
using FluentValidation;

namespace ShotcareApp.Validation
{
    public class NewFacilityValidator : AbstractValidator<ShotcareApp.Models.ShotEntry.Facility>
    {
        public NewFacilityValidator()
        {
            RuleFor(p => p.Name).NotEmpty().WithMessage("Please enter a Facility Name");
            RuleFor(p => p.MaximumStudents).NotEmpty().WithMessage("Please enter maximum number of students");
        }
    }
}
```
Thanks

**Comment by chrissainty** [comment]
Yes, it's possible. The code you have looks correct. Are you experiencing any errors?

**Reply by TrailBlazor** [reply]
No errors, it just ignores the validation. It works if I  don't DisableAssemblyScanning and allow it to be matched by the model, but of course that defeats the purpose.

Have you tried it?

Thanks,
Dave

**Reply by TrailBlazor** [reply]
Hi @chrissainty 
It looks like I have to add this to the Startup but, of course, this defeats the purpose since I want to have different validators for the same model depending on what page they are on.
`            services.AddTransient<IValidator<ShotcareApp.Models.ShotEntry.Facility>, NewFacilityValidator>();
`
Is there a way to specify different validators for the same model?

Thanks,
Dave
